### PR TITLE
fix: remove potential runtime panic in x/feegrant

### PR DESCRIPTION
**Pull request by ulbqb**
## Description

I added safe type casting and then removed potential runtime panic.

## Motivation and context

Unsafe type casting may cause potential runtime panic.

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/line/lbm-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/line/lbm-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

File: x/feegrant/filtered_fee_test.go [hunk 1]
```diff
@@ -188,3 +189,80 @@ func TestFilteredFeeValidAllow(t *testing.T) {
...
+func (i invalidAllowance) Accept(ctx sdk.Context, fee sdk.Coins, msgs []sdk.Msg) (remove bool, err error) {
+	return false, nil
+}
+
+func (i invalidAllowance) ValidateBasic() error {
+	return nil
+}
+
+// invalidAllowance2 does not have field
+type invalidAllowance2 struct {
```
Comment (0Tech) [inline x/feegrant/filtered_fee_test.go:209]: You may make it to extend `invalidAllowance1`.

**Review by 0Tech — APPROVED** [review]
It would be better to provide more meaningful names to the mocks.

File: x/feegrant/filtered_fee_test.go (hunk 1 above)
Comment (tnasu) [reply x/feegrant/filtered_fee_test.go:209]: And I prefer to have specific naming. How about `invalidFieldAllowance`?

Comment (0Tech) [reply x/feegrant/filtered_fee_test.go:209]: Is the absence of the field relevant to the failure? It seems like, being `nil` is the trigger, isn't it?

Comment (ulbqb) [reply x/feegrant/filtered_fee_test.go:209]: Yes, `nil` is the trigger. I fix the test case name.

File: x/feegrant/filtered_fee_test.go [hunk 2]
```diff
@@ -188,3 +189,73 @@ func TestFilteredFeeValidAllow(t *testing.T) {
...
+		"valid allowance": {
+			allowance: &feegrant.BasicAllowance{},
+			valid:     true,
+		},
+		"invalid interface allowance": {
+			allowance: &invalidInterfaceAllowance{},
+			valid:     false,
+		},
+		"empty allowance": {
+			allowance: (*invalidProtoAllowance)(nil),
```
Comment (0Tech) [inline x/feegrant/filtered_fee_test.go:241]: Do you really need `invalidProtoAllowance`? Is this possible:
```suggestion
			allowance: (*feegrant.BasicAllowance)(nil),
```

Comment (ulbqb) [reply x/feegrant/filtered_fee_test.go:241]: That change cause panic at [`proto.Marshal`](https://github.com/line/lbm-sdk/blob/d48d8a7094acb0253e0ba9b279f4ee5de3daef4a/codec/types/any.go#L68).
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
```

**Review by zemyblue — APPROVED** [review]

**Pull request merged**